src/main/java/uk/co/real_logic/benchmarks/rtt/InMemoryMessageTransceiver.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2015-2020 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.co.real_logic.benchmarks.rtt;
+
+import static java.lang.Integer.numberOfLeadingZeros;
+import static java.util.Arrays.fill;
+import static org.agrona.BitUtil.CACHE_LINE_LENGTH;
+import static org.agrona.BitUtil.SIZE_OF_LONG;
+import static org.agrona.UnsafeAccess.UNSAFE;
+import static sun.misc.Unsafe.ARRAY_LONG_BASE_OFFSET;
+import static sun.misc.Unsafe.ARRAY_LONG_INDEX_SCALE;
+
+abstract class InMemoryMessageTransceiver1 extends MessageTransceiver
+{
+    static final int SIZE = 4096;
+    private static final int MASK = SIZE - 1;
+    private static final int SHIFT = 31 - numberOfLeadingZeros(ARRAY_LONG_INDEX_SCALE);
+    private static final int PADDING_OFFSET = CACHE_LINE_LENGTH / SIZE_OF_LONG;
+    final long[] messages = new long[SIZE];
+
+    // Padding
```

Comment:
Why pad with booleans rather than longs?

Reply:
Basically, because of this https://twitter.com/chvest/status/1232594808214097921

Reply:
Interesting. We'll need to take account of this in Agrona and Aeron.

Reply:
Sure